[chengjunyuan] iP

chengjunyuan opened the pull request:
### Gronk
Gronk is a simple bot designed to help you keep track of your tasks and deadlines 🗿 Here's a quote from a reviewer:
> Looks good to me!
> -- jskimdev

Commands Supported:
- List
- Add
- Delete
- Find
- Say `bye` to exit the bot and save your tasks to an external file!

Features this blurb has:
- [X] Heading
- [X] Bullet List
- [X] Numbered List
- [X] Fenced Code Block
- [X] Task List
- [X] Emoji
- [X] Blockquote
- [X] A hyperlink
- [X] Inline code
- [X] Some text formatting, e.g., **bold** and _italics_

Gronk is currently undergoing unit testing with [JUnit](https://junit.org/junit5/docs/current/user-guide/), for example:
```java
    @Test
    public void listTest(){
        Storage storage = new Storage("testTasks.txt");
        storage.loadFromFile();
        Parser parser = new Parser(storage.returnTasks());
        String expected = "\t1. [T] [ ] gaming\n"
                + "\t2. [D] [ ] swim (by: Jan 1 2025)\n"
                + "\t3. [T] [X] game\n"
                + "\t4. [T] [X] fight";
        assertEquals(parser.parse("list"), expected);
    }
```

Upcoming Features / Work:
1. General code refactoring
2. 100% testing coverage
3. Fix all known bugs (there are some 😢 )

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

chengjunyuan committed:
Level-0

chengjunyuan committed:
Level-1

chengjunyuan committed:
Level-2

chengjunyuan committed:
Level-3

chengjunyuan committed:
Level-4 WIP.

chengjunyuan committed:
Level-4 WIP. Moved some code around to make main cleaner. Still need to work on fixing the printed messages (there are some problems with spaces and whatnot)!

chengjunyuan committed:
Level-4 done!

chengjunyuan committed:
A-TextUiTesting

chengjunyuan committed:
Level-5

chengjunyuan committed:
Level-6

chengjunyuan committed:
General refactoring, testing of file I/O

chengjunyuan committed:
Add Gronk.java

chengjunyuan committed:
Add datetime functionality to Deadline and Event.

jskimdev commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,46 @@
+        return this.desc;
+    }
+
+    public String statusMessage() {
```
The naming for methods needs to be verb. For instance, getName().
```suggestion
    public String getStatusMessage() {
```
```suggestion
    public String returnStatusMessage() {
```

chengjunyuan replied:
Thanks for the feedback, have taken it into account when I finished some major refactoring today. Will continue on ironing out the issues.

jskimdev commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,46 @@
+public class Task {
```
Javadocs is not present in Class and Methods.

```java
/**
 * Returns lateral location of the specified position.
 * If the position is unset, NaN is returned.
 *
 * @param x X coordinate of position.
 * @param y Y coordinate of position.
 * @param zone Zone of position.
 * @return Lateral location.
 * @throws IllegalArgumentException  If zone is <= 0.
 */
public double computeLocation(double x, double y, int zone)
        throws IllegalArgumentException {
    //...
}
```

chengjunyuan replied:
Thanks for the feedback, have added JavaDoc for roughly half of the non-private methods. Will continue working towards 100% JavaDoc coverage!

jskimdev commented on src/main/java/Gronk.java:
```diff
@@ -0,0 +1,174 @@
+    }
+
+    public static void loadMessage(String m) {
+        String[] message = m.split(",");
```
Plural form should be used on names representing a collection of objects. I am not sure about this, but seems like lists need to have plural forms.

chengjunyuan replied:
You're corrected, have resolved this in the latest round of refactoring. Thanks for the spot!

jskimdev commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,46 @@
+        this.status = s;
+    }
+
+    public void update(int i) {
```
maybe the name for this method can be setStatus? Just a suggestion.

chengjunyuan replied:
Good suggestion, will implement. Thank you!

jskimdev commented on src/main/java/Gronk.java:
```diff
@@ -0,0 +1,174 @@
+import java.io.FileNotFoundException;
+
+public class Gronk {
+    public static String WELCOME = "\tHi, I'm Gronk!\n"
+            + "\tWhat are we up to today?";
+    public static String GOODBYE = "\tSystem closing. Goodbye!";
```
Constants are static final dataType. If they are not meant to be constants, need to follow camelCase (welcome and goodbye).

chengjunyuan replied:
Have updated to make them `public static final`, thanks for the catch!

jskimdev commented on text-ui-test/input.txt:
Good to see many testcases.

jskimdev reviewed (commented):
Looks good to me! I guess you haven't finished the week3 tasks yet, so Javadocs would be done when you finish week3 tasks. Overall, just some mistakes in naming, but looks good. Keep up the good work!

azumieflare commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,62 @@
+    //}
+
+    @Override
+    public String statusMessage() {
```
Could be renamed to better reflect what this method does.
For example: ```getStatusMessage() {```

chengjunyuan replied:
Will implement, thank you for the suggestion!

azumieflare commented on src/main/java/Gronk.java:
```diff
@@ -0,0 +1,174 @@
+    public static String GOODBYE = "\tSystem closing. Goodbye!";
+    public static ArrayList<Task> tasks;
+
+    public static void lines() { // prints out a line
```
Could be renamed to better reflect what this method does.
For example: ```printLine() {```

chengjunyuan replied:
Thanks for the suggestion, this has been fixed in the latest refactoring.

azumieflare commented on src/main/java/Gronk.java:
```diff
@@ -0,0 +1,174 @@
+import java.io.FileNotFoundException;
+
+public class Gronk {
+    public static String WELCOME = "\tHi, I'm Gronk!\n"
+            + "\tWhat are we up to today?";
+    public static String GOODBYE = "\tSystem closing. Goodbye!";
```
Perhaps can leave these variables as ```public static final``` as they should be fixed and not changed at all?

chengjunyuan replied:
Good catch, have fixed this issue.

azumieflare reviewed (commented):
Overall looks good! Just need to fix the consistency with method names.

chengjunyuan committed:
Add more OOP

chengjunyuan committed:
Add package

chengjunyuan committed:
Add package

chengjunyuan committed:
Merge branch 'add-gradle-support' of ssh://github.com/chengjunyuan/ip

chengjunyuan committed:
Add JUnit tests

chengjunyuan committed:
Add .jar

chengjunyuan committed:
Add JavaDoc for main classes.

chengjunyuan committed:
Refactor to satisfy coding standard

chengjunyuan committed:
Add find function

chengjunyuan committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDoc for main classes.

chengjunyuan committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add find function

chengjunyuan committed:
Refactor to comply with Checkstyle

chengjunyuan committed:
Checkstyle files

chengjunyuan committed:
Refactor Parser (I am thinking about it)

chengjunyuan committed:
Refactor Parser class.

Main Parser method is split into multiple
smaller Parser methods. This is done to shorten
the main Parser method from 130+ lines to
just over 30 lines. All code logic of the
Parser remains the same.

chengjunyuan committed:
Merge pull request #1 from chengjunyuan/branch-A-CodeQuality

branch-A-CodeQuality

chengjunyuan committed:
Major refactor of parser

chengjunyuan committed:
Edit testTasks to remove old format of deadline

chengjunyuan committed:
Add A-Varargs to Parser

chengjunyuan committed:
Level-10

chengjunyuan committed:
Testing GUI

chengjunyuan committed:
General refactor according to checkstyle comments

chengjunyuan committed:
Minor refactor

chengjunyuan committed:
Add assert statements

Make sure that the methods are being called with a correctly formatted String.

chengjunyuan committed:
Merge pull request #3 from chengjunyuan/branch-A-Assertions

Add assert statements

chengjunyuan committed:
Add BCD-Extension

The following extensions were added:
- C-Help
- C-MassOps
- C-BetterSearch

Documentation coming soon!

chengjunyuan committed:
Rename Duke instances to Gronk

chengjunyuan committed:
Update documentation with screenshots and features

chengjunyuan committed:
Update help function in Parser

chengjunyuan committed:
Final refactoring
